### ISSUE TITLE
Accessibility: Color contrast between states insufficient [Account Details]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51836]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The icon to show and hide the password goes from black to very dark gray when activated, with a color contrast ratio of 2.95:1 between the two states.

This is a low priority issue because the actual visibility of the password also communicates the state of the button.

![T007_KB_ShowHideIconMissingRoleName](https://github.com/user-attachments/assets/462b6384-46c5-478b-8f43-2076744b1bdc)

### Component or Page: 

Account details My Account Page

### Name affected element:

Show/Password icon

### Relevant code snippets:

`.woocommerce form .show-password-input.display-password::after, .woocommerce-page form .show-password-input.display-password::after {
    color: #585858;
}`

### Recommended fix:

Increase the color contrast between the states to have a color contrast ratio betwene the states of at least 3:1. The show/hide password button may be enhanced by also changing the visual appearance of the icon, such as adding and removing a line through the eye.

**Comment by darrenjacoby** [comment]
@dkotter Noting that I was not able to replicate this on any of the newer themes, however I have added a PR to update the legacy code.

cc @vikrampm1
